src/Microsoft.Azure.ServiceBus/MessageSession.cs

```diff
@@ -1,41 +1,32 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.Azure.ServiceBus.Amqp
+namespace Microsoft.Azure.ServiceBus
 {
     using System;
-    using System.Collections.Generic;
     using System.IO;
+    using System.Threading;
     using System.Threading.Tasks;
-    using Microsoft.Azure.Amqp;
-    using Microsoft.Azure.ServiceBus.Core;
+    using Amqp;
+    using Azure.Amqp;
+    using Core;
 
     internal class MessageSession : MessageReceiver, IMessageSession
     {
-        public MessageSession(string sessionId, DateTime lockedUntilUtc, MessageReceiver innerMessageReceiver, RetryPolicy retryPolicy)
-            : base(innerMessageReceiver.ReceiveMode, innerMessageReceiver.OperationTimeout, retryPolicy)
-        {
-            this.SessionId = sessionId;
-            this.LockedUntilUtc = lockedUntilUtc;
-            this.InnerMessageReceiver = innerMessageReceiver ?? throw Fx.Exception.ArgumentNull(nameof(innerMessageReceiver));
-            this.ReceiveMode = innerMessageReceiver.ReceiveMode;
-        }
-
-        public override string Path
-        {
-            get { return this.InnerMessageReceiver.Path; }
-        }
-
-        protected MessageReceiver InnerMessageReceiver { get; set; }
-
-        protected override async Task OnClosingAsync()
-        {
-            if (this.InnerMessageReceiver != null)
-            {
-                await this.InnerMessageReceiver.CloseAsync().ConfigureAwait(false);
-            }
-        }
-
+        public MessageSession(
+            string entityPath,
+            MessagingEntityType? entityType,
+            ReceiveMode receiveMode,
+            ServiceBusConnection serviceBusConnection,
+            ICbsTokenProvider cbsTokenProvider,
+            RetryPolicy retryPolicy,
+            int prefetchCount = 0,
```

Comment:
Should this be using `Constants`? Similar to how a default is defined for [message pump](https://github.com/Azure/azure-service-bus-dotnet/blob/7c59009fe5b96f533d10d27fc3a24322d386f342/src/Microsoft.Azure.ServiceBus/Constants.cs#L40).